saleor/plugins/sendgrid/plugin.py

```diff
@@ -0,0 +1,255 @@
+            "account_change_email_request_template_id"
+        ),
+        UserNotifyEvent.ACCOUNT_PASSWORD_RESET: "account_password_reset_template_id",
+        UserNotifyEvent.INVOICE_READY: "invoice_ready_template_id",
+        UserNotifyEvent.ORDER_CONFIRMATION: "order_confirmation_template_id",
+        UserNotifyEvent.ORDER_CONFIRMED: "order_confirmed_template_id",
+        UserNotifyEvent.ORDER_FULFILLMENT_CONFIRMATION: (
+            "order_fulfillment_confirmation_template_id"
+        ),
+        UserNotifyEvent.ORDER_FULFILLMENT_UPDATE: (
+            "order_fulfillment_update_template_id"
+        ),
+        UserNotifyEvent.ORDER_PAYMENT_CONFIRMATION: (
+            "order_payment_confirmation_template_id"
+        ),
+        UserNotifyEvent.ORDER_CANCELED: "order_canceled_template_id",
+        UserNotifyEvent.ORDER_REFUND_CONFIRMATION: (
+            "order_refund_confirmation_template_id"
+        ),
+    }
+
+
+class SendgridEmailPlugin(BasePlugin):
+    PLUGIN_ID = "mirumee.notifications.sendgrid_email"
+    PLUGIN_NAME = "Sendgrid"
+    DEFAULT_ACTIVE = False
+
+    DEFAULT_CONFIGURATION = [
+        {"name": "sender_name", "value": ""},
+        {"name": "sender_address", "value": ""},
+        {"name": "account_confirmation_template_id", "value": None},
+        {"name": "account_set_customer_password_template_id", "value": None},
+        {"name": "account_delete_template_id", "value": None},
+        {"name": "account_change_email_confirm_template_id", "value": None},
+        {"name": "account_change_email_request_template_id", "value": None},
+        {"name": "account_password_reset_template_id", "value": None},
+        {"name": "invoice_ready_template_id", "value": None},
+        {"name": "order_confirmation_template_id", "value": None},
+        {"name": "order_confirmed_template_id", "value": None},
+        {"name": "order_fulfillment_confirmation_template_id", "value": None},
+        {"name": "order_fulfillment_update_template_id", "value": None},
+        {"name": "order_payment_confirmation_template_id", "value": None},
+        {"name": "order_canceled_template_id", "value": None},
+        {"name": "order_refund_confirmation_template_id", "value": None},
+        {"name": "api_key", "value": None},
+    ]
+    CONFIG_STRUCTURE = {
+        "sender_name": {
+            "type": ConfigurationTypeField.STRING,
+            "help_text": "Name which will be visible as 'from' name.",
+            "label": "Sender name",
+        },
+        "sender_address": {
+            "type": ConfigurationTypeField.STRING,
+            "help_text": "Sender email which will be visible as 'from' email.",
+            "label": "Sender email",
+        },
+        "account_confirmation_template_id": {
+            "type": ConfigurationTypeField.STRING,
+            "help_text": "The dynamic template ID for account confirmation.",
+            "label": "Template ID for account confirmation",
```

Comment:
The text is duplicated in both fields, maybe instead in "label" we could name the template and in "help text" describe in what form the template should be provided, something like this:

label: "Account confirmation email template"
help text: "ID of the dynamic template in Sendgrid"